File: app/src/main/java/com/practice/todo/storage/database/dao/TodoItemDao.java
package com.practice.todo.storage.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.practice.todo.storage.database.entity.TodoItem;

/**
 * TodoItem数据库操作类
 */
@Dao
public interface TodoItemDao {
    @Query("SELECT * FROM todo_item")
    TodoItem[] loadAll();
    @Query("SELECT * FROM todo_item WHERE id = :id")
    TodoItem loadById(int id);
    @Insert
    void insert(TodoItem item);
    @Update
    void update(TodoItem... items);
    @Delete
    void delete(TodoItem... items);
}

File: app/src/main/java/com/practice/todo/storage/database/dao/TodoItemDao.kt
package com.practice.todo.storage.database.dao

import androidx.room.*
import com.practice.todo.storage.database.entity.TodoItem

@Dao
interface TodoItemDao {

    @Query("SELECT * FROM todo_item")
    fun loadAll(): Array<TodoItem>

    @Query("SELECT * FROM todo_item WHERE id = :id")
    fun loadById(id: Int): TodoItem

    @Insert
    fun insert(todoItem: TodoItem)

    @Update
    fun update(vararg todoItems: TodoItem)

    @Delete
    fun delete(vararg todoItems: TodoItem)
}